WebDriverAgentLib/Commands/FBOrientationCommands.m

```diff
@@ -84,16 +101,31 @@ + (BOOL)setDeviceOrientation:(NSString *)orientation forApplication:(FBApplicati
     return NO;
   }
   [XCUIDevice sharedDevice].orientation = orientationValue.integerValue;
+  return [self waitUntilApplication:application isOrientation:orientationValue.integerValue];
+}
 
-  // We have a busy loop here while we wait for the orientation to change as we do not have any hooks
-  // into the event being handled.
-  // If we could just hook into the event handler to know when it has been processed..
-  NSDate *startDate = [NSDate date];
-  while (![[self interfaceOrientationForApplication:application] isEqualToString:orientation] && (-1 * [startDate timeIntervalSinceNow]) < kFBWebDriverOrientationChangeDelay) {
-    CFRunLoopRunInMode(kCFRunLoopDefaultMode, 0.0, YES);
-  }
++ (BOOL)setDeviceRotation:(NSDictionary *)rotationObj forApplication:(FBApplication *)application
```

Comment:
Looks good to me. However could you move those methods to XCUIDevice category and write an integration test for it?